client/jetpack-cloud/sections/site-landing.jsx

```diff
@@ -8,22 +8,37 @@ import {
 	isFetchingPartner,
 	isAgencyUser,
 } from 'calypso/state/partner-portal/partner/selectors';
+import { getPreference } from 'calypso/state/preferences/selectors';
+import { JETPACK_AGENCY_DASHBOARD_DEFAULT_FILTER_CLEARED_KEY } from './agency-dashboard/sites-overview/utils';
+
 import '../style.scss';
 
 export default function SiteLanding( { primarySiteSlug, isPrimarySiteJetpackSite } ) {
 	const hasFetched = useSelector( hasFetchedPartner );
 	const isFetching = useSelector( isFetchingPartner );
+	const isDefaultFilterCleared = useSelector( ( state ) =>
+		getPreference( state, JETPACK_AGENCY_DASHBOARD_DEFAULT_FILTER_CLEARED_KEY )
+	);
 	const isAgency = useSelector( isAgencyUser );
 	const isAgencyEnabled = config.isEnabled( 'jetpack/agency-dashboard' );
 	useEffect( () => {
 		if ( hasFetched ) {
 			if ( isAgency && isAgencyEnabled ) {
-				page.redirect( '/dashboard' );
+				page.redirect(
+					isDefaultFilterCleared ? '/dashboard' : '/dashboard?issue_types=all_issues'
+				);
 				return;
```

Comment:
I feel we're repeating ourselves too much here (I see four places where we do the same computation). Would it be possible to abstract this logic into its own function? It will make updating the logic way easier in the future.

Reply:
Yes, totally makes sense 👍